File: wlr/OGUR/OGUR/Entities/Extensions.cs
using OGUR.Creatures;
using SPX.Entities;
using SPX.Core;
using System.Collections.Generic;

namespace OGUR.Entities
{
    public static class Extensions
    {
        public static ICreature IsCreature(this IEntity entity)
        {
            if (entity.GetEntityType() == OGUR.EntityType.ACTOR)
            {
                return entity as ICreature;
            }
            return null;
        }
    }
}

File: port-workspace/aigilas-ported/src/com/aigilas/entities/Extensions.java
package com.aigilas.entities;import com.xna.wrapper.*;import java.util.*;import com.aigilas.creatures.*;import com.spx.entities.*;import com.spx.core.*;
    public class Extensions
    {
        public static ICreature IsCreature(IEntity entity)
        {
            if (entity.GetEntityType() == com.aigilas.EntityType.ACTOR)
            {
                return (ICreature)entity;
            }
            return null;
        }
    }
